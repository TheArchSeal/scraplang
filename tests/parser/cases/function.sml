fn f() {}
fn f(x: a, y: b = 0): c { 0; }
