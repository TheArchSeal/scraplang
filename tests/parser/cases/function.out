stmt (11):1:1 fn f
    type (1):1:1 (inferred)
    stmt (2):1:9 {}
stmt (11):2:1 fn f
    param   :2:6 x
    type (2):2:9 a
    expr (1):2:6 (empty)
    param   :2:12 y
    type (2):2:15 b
    expr (2):2:19 0
    type (2):2:23 c
    stmt (2):2:27 {}
        stmt (3):2:27 ;
            expr (2):2:27 0
